src/Riok.Mapperly/SymbolExtensions.cs

```diff
@@ -0,0 +1,62 @@
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis;
+
+public static class SymbolExtensions
+{
+    public static bool IsPartialDefinition(this IMethodSymbol method)
+    {
+#if ROSLYN3_10_OR_GREATER
+        return method.IsPartialDefinition;
+#else
+        if (method.Locations.Any(loc => loc.IsInMetadata))
+        {
+            return false;
+        }
+
+        var isPartial = method.DeclaringSyntaxReferences
+            .Select(r => r.GetSyntax())
+            .Where(s => s.IsKind(SyntaxKind.MethodDeclaration))
+            .Cast<MethodDeclarationSyntax>()
+            .Any(s => s.Modifiers.Any(SyntaxKind.PartialKeyword));
+        return isPartial && method.PartialImplementationPart == null && method.PartialDefinitionPart == null;
```

Comment:
Why does the definition part need to be null? Shouldn't it be not null?

Reply:
This is the wierd part, see: https://github.com/dotnet/roslyn/issues/51082
Will add more info in comment